hack/generated/pkg/genruntime/conditions/conditions.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package conditions
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: Hah, name...
+type Conditioner interface {
```

Comment:
Yeah, not a fan of this name either. 

But the best I've got to offer is `ConditionProvider` and `ConditionCointainer`, neither of which is better.

Reply:
Cluster API does:
```
type Setter interface {
	Getter
	SetConditions(clusterv1.Conditions)
}
```

So it's used as `conditions.Setter`, which is maybe not bad? I didn't split the interface into `Get` and `Set`, but I could.